### PR TITLE
[avmfritz] Avoid a hidden NPE when getIdentifier() of an uninitialized ThingHandler is called

### PR DESCRIPTION
**Pull request by cweitkamp**
- Avoid a hidden NPE when `getIdentifier()` of an uninitialized `ThingHandler` is called

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

**Comment by cweitkamp** [comment]
Can be backported to OH 2.5.x, if possible.

File: bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/handler/AVMFritzBaseBridgeHandler.java [hunk 1]
```diff
@@ -270,11 +269,10 @@ public void onDeviceListAdded(List<AVMFritzBaseModel> deviceList) {
                 .collect(Collectors.toMap(it -> it.getIdentifier(), Function.identity()));
         getThing().getThings().forEach(childThing -> {
             final AVMFritzBaseThingHandler childHandler = (AVMFritzBaseThingHandler) childThing.getHandler();
+            // if (childHandler != null && ThingHandlerHelper.isHandlerInitialized(childHandler)) {
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/handler/AVMFritzBaseBridgeHandler.java:272]: Is this line expected to be here?

Comment (cweitkamp) [reply bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/handler/AVMFritzBaseBridgeHandler.java]: No, that was my backup plan ... removed.

**Review by Bushido76 — APPROVED** [review]
The hidden NPE was initially traced down and reported by me as it lead to various instabilities in Thing handling. The correction was discussed with Christoph and applied by him. The binding - with all corrections applied - is running on my local OH3 instance since yesterday noon and the binding is behaving much more reliably since.

**Review by kaikreuzer — APPROVED** [review]

**Comment by kaikreuzer** [comment]
> Can be backported to OH 2.5.x, if possible.

I'm afraid cherry-picking results in a merge conflict in the imports of the Powerline handler class.
You'll have to do it locally, resolve the conflict and create a dedicated PR for that, sorry.

**Comment by cweitkamp** [comment]
Okay, see https://github.com/openhab/openhab-addons/pull/9043.

**Pull request merged**